apps/Level.py

```diff
@@ -0,0 +1,119 @@
+        b'B@\xa8N\x10N\x80r\x82\xcf\xc0\xfd\xc1@\xb0C'
+        b'\x02\xc1\x04\xc1\x02C\xc1O\x82\x80\xa8\x8e\x10\x8e\xc0r'
+        b'\xc2O@\xfdA\x80\xb0\x85A\x84A\x85A\x8f\xc2\xc0'
+        b'\xa8\xce\x10\xce@rB\x8f\x80\xfd\x81\xc0\xb0\xc5\x81\xc4'
+        b'\x81\xc5\x81\xcfB@\xa8N\x10N\x80r\x82\xcf\xc0\xfd'
+        b'\xc1@\xb0E\xc1D\xc1E\xc1O\x82\x80\xa8\x8e\x10\x8e'
+        b'\xc0r\xc2O@\xfdA\x80\xb0\x85A\x84A\x85A\x8f'
+        b'\xc2\xc0\xa8\xce\x10\xce@rC\x8e\x80\xfd\x81\xc0\xb0\xc5'
+        b'\x81\xc4\x81\xc5\x81\xceC@\xa8N\x10O\x80r\x82\xce'
+        b'\xc0\xfd\xc1@\xb0E\xc1D\xc1E\xc1N\x82\x80\xa8\x8f'
+        b'\x10\x8f\xc0r\xc3M@\xfdA\x80\xb0\x85A\x84A\x85'
+        b'A\x8d\xc3\xc0\xa8\xcf\x10\xd0@rD\x8b\x80\xfd\x81\xc0'
+        b'\xb0\xc5\x81\xc4\x81\xc5\x81\xcbD@\xa8P\x10Q\x80r'
+        b'\xaeQ\x10S\xaaS\x10\x7f\x11\x10\x7f\x11\x10\x7f\x11\x10'
+        b'\x7f\x11\x10\x7f\x11\x10\x7f\x11\x10\x7f\x11\x10\x7f\x11\x10\x7f'
+        b'\x11\x10\x7f\x11\x10\x7f\x11\x10\x7f\x11\x11\x7f\x0f\x13\x7f\r'
+        b'?\xff\xff\xff\xff\xff\xd0'
+    )
+
+    def __init__(self):
+        self.old_xy = (0,0)
+        self.calibration = (0,0)
+        self.prompt = False
+        self.calibrate = widgets.Button(20, 20, 200, 60, 'Calibrate')
+        self.reset = widgets.Button(20, 90, 200, 60, 'Reset')
+        self.cancel = widgets.Button(20, 160, 200, 60, 'Cancel')
+
+    def foreground(self):
+        self.prompt = False # in case the watch went to sleep with prompt on
+        self._draw()
+        wasp.system.request_event(wasp.EventMask.TOUCH)
+        wasp.system.request_tick(125)
+
+    def _draw(self):
+        wasp.watch.drawable.fill()
+        self._update()
+
+    def _update(self):
+        if not self.prompt:
+            draw = wasp.watch.drawable
+            draw.fill(None, self.old_xy[0] - 3 + _X_CENTER, self.old_xy[1] - 3 + _Y_CENTER, 6, 6)
+            # draw guide lines
+            draw.line(0, _Y_CENTER, _X_MAX, _Y_CENTER, color = wasp.system.theme('mid'))
+            draw.line(_X_CENTER, 0, _X_CENTER, _Y_MAX, color = wasp.system.theme('mid'))
+            # We save x as y and -y as x because we use the screen's coordinate
+            # system.
+            # We also clamp and scale the values down a bit to make them fit better,
+            # and apply the calibration
+            (new_y, new_x, _) = watch.accel.accel_xyz()
+            new_x = min(_X_CENTER, max(-_X_CENTER, (new_x-self.calibration[0])//-3))
+            new_y = min(_Y_CENTER, max(-_Y_CENTER, (new_y-self.calibration[1])//3))
+            draw.fill(wasp.system.theme('bright'), new_x - 3 + _X_CENTER, new_y - 3 + _Y_CENTER, 6, 6)
+            self.old_xy = (new_x, new_y)
+
+    def tick(self, ticks):
+        self._update()
+        wasp.system.keep_awake()
+
+    def touch(self, event):
+        if self.prompt:
+            # Handle buttons
+            if self.calibrate.touch(event):
+                (y, x, _) = watch.accel.accel_xyz()
```

Comment:
Like #277 this means that the orientation of the accelerometer ends up burned into the apps.

Ideally the driver layer would handle the remap so that the x/y on the accelerometer would match x/y on the display (in other words the x value should be positive if the left side of the watch frame is higher than the right).

It's actually a little awkward to solve though since the BMA421 driver can't know what orientation the unit is within the watch frame either so the BMA421 driver would need to be instanciated with a remapping value (I suspect a single 32-bit value is more than enough to describe the mapping and sign between sensor x/y/z and frame x/y/z).